components/org.wso2.carbon.identity.fraud.detection.sift/src/main/java/org/wso2/carbon/identity/fraud/detection/sift/util/Util.java

```diff
@@ -0,0 +1,284 @@
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.fraud.detection.sift.Constants;
+import org.wso2.carbon.identity.fraud.detection.sift.internal.SiftDataHolder;
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.governance.bean.ConnectorConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.CONNECTOR_NAME;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.HTTP_SERVLET_REQUEST;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.LOGIN_TYPE;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.SIFT_API_KEY_PROP;
+import static org.wso2.carbon.identity.fraud.detection.sift.Constants.USER_AGENT_HEADER;
+
+/**
+ * Util class to build the payload to be sent to Sift.
+ */
+public class Util {
+
+    private static final Log LOG = LogFactory.getLog(Util.class);
+
+    public static JSONObject buildPayload(JsAuthenticationContext context, String loginStatus,
+                                          Map<String, Object> passedCustomParams)
+            throws FrameworkException {
+
+        String loginSts = getLoginStatus(loginStatus).getSiftValue();
+
+
+        JSONObject payload = new JSONObject();
+
+        // Add the required parameters to the payload.
+        payload.put(Constants.TYPE, LOGIN_TYPE);
+        payload.put(Constants.API_KEY, getSiftApiKey(context.getWrapped().getTenantDomain()));
+        payload.put(Constants.LOGIN_STATUS, loginSts);
+        payload.put(Constants.USER_ID_KEY, resolvePayloadData(Constants.USER_ID_KEY, context));
+
+        Map<String, String> browserProperties = new HashMap<>();
+        browserProperties.put(Constants.USER_AGENT_KEY, resolvePayloadData(Constants.USER_AGENT_KEY, context));
+        payload.put(Constants.BROWSER_KEY, browserProperties);
+
+        payload.put(Constants.IP_KEY, resolvePayloadData(Constants.IP_KEY, context));
+
+        payload.put(Constants.SESSION_ID_KEY, resolvePayloadData(Constants.SESSION_ID_KEY, context));
+
+        processDefaultParameters(passedCustomParams, payload);
+
+        if (passedCustomParams != null) {
+            for (Map.Entry<String, Object> entry : passedCustomParams.entrySet()) {
+                payload.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return payload;
+    }
+
+    // Process the default parameters and remove them from the custom parameters. If the default parameters are
+    // passed as empty values, remove them from the payload.
+    private static void processDefaultParameters(Map<String, Object> passedCustomParams, JSONObject payload) {
+
+        if (passedCustomParams != null) {
```

Comment:
Let remove the if nesting.

```suggestion
        if (passedCustomParams == null) { 
           return;
        }
```